spinalcordtoolbox/scripts/sct_warp_template.py

```diff
@@ -120,20 +116,22 @@ def warp_label(path_label, folder_label, file_label, fname_src, fname_transfo, p
                                     '-d', fname_src,
                                     '-w', fname_transfo,
                                     '-o', os.path.join(path_out, folder_label, template_label_file[i]),
-                                    '-x', get_interp(template_label_file[i], list_labels_nn),
+                                    '-x', get_interp(template_label_file[i]),
                                     '-v', '0'])
         # Copy list.txt
         copy(os.path.join(path_label, folder_label, file_label), os.path.join(path_out, folder_label))
 
 
 # Get interpolation method
 # ==========================================================================================
-def get_interp(file_label, list_labels_nn):
+def get_interp(file_label):
     # default interp
     interp = 'linear'
     # Nearest Neighbours interp
-    if any(substring in file_label for substring in list_labels_nn):
+    if any(substring in file_label for substring in ['_levels.', '_csf.', '_cord.']):
         interp = 'nn'
+    elif any(substring in file_label for substring in ['_label_', '_midpoint.']):
```

Comment:
Thank you for making this change! I agree with the reasoning outlined in commit https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4281/commits/c97eb5e355e23e01bcf3e868847d056d6328d505.

```suggestion
    # For safety and consistency, ensure strings are bracketed by `_` or `.` on both sides
    if any(substring in file_label for substring in ['_levels.', '_csf.', '_cord.']):
        interp = 'nn'
    elif any(substring in file_label for substring in ['_label_', '_midpoint.']):
```

Reply:
I also quickly manually double-checked this by setting breakpoints within the `elif` cases, then ensuring that they were triggered appropriately. This all works well!